### PR TITLE
Data agg2

### PR DESCRIPTION
**Pull request by bdyetton**
Added some parsing of 2 variable data to data frame

### CODE REVIEW

File: sharepa/analysis.py [hunk 1]
```diff
@@ -16,6 +16,33 @@ def bucket_to_dataframe(name, bucket):
     ).T
 
 
+def convert_key(x):
+    x.key = x.key_as_string
+    return x
+
+
+def aggregation_to_dataframe(name,agg,filter=None):
```
Comment (fabianvf) [inline sharepa/analysis.py:24]: `filter` is actually a python function, so you should call this variable something else. Also, would it be possible to take the Response.aggregations and find all name-bucket pairs?

**Comment by fabianvf** [comment]
Also, adding some tests would be nice. You can mock the return values of functions (such as `.execute()`) using the `mock` package.

File: sharepa/analysis.py (hunk 1 above)
Comment (bdyetton) [inline sharepa/analysis.py:24]: true that. Yep, good idea, ill give that a go

Regards,
Ben Yetton

Graduate Student,
Mednick Sleep and Cognition Lab,
University Of California, Riverside
900 University Ave
Riverside, CA 92521
benjamin.yetton@email.ucr.edu

Comment (fabianvf) [inline sharepa/analysis.py:24]: Also, arguments should have a space after the comma `def x(p1, p2, p3)`

File: sharepa/analysis.py [hunk 2]
```diff
@@ -16,6 +16,33 @@ def bucket_to_dataframe(name, bucket):
...
+        :type name: str
+        :param bucket: a bucket from elasticsearch results
+        :type bucket: list[dict]
+        :returns: pandas.DataFrame
+    '''
+    if hasattr(agg, 'buckets'):
+        if filter is not None:
+            expanded_agg = []
+            for bucket in agg.buckets:
+                expanded_agg.append(bucket_to_dataframe(bucket.key, map(convert_key, getattr(bucket,filter).buckets)))
```
Comment (fabianvf) [inline sharepa/analysis.py:37]: Not all aggregations will have the `key_as_string` key. Might be better to make dataframes that are just indexed by row number, and make each key in the bucket it's own column, perhaps namespaced by the name. So like:

|  | dates.key | dates.key_as_string | dates.<other_thing> |
| --- | --- | --- | --- |
| 0 | key0 | key_as_string0 | <other_thing>0 |
| 1 | key1 | key_as_string1 | <other_thing>1 |

Comment (fabianvf) [inline sharepa/analysis.py:37]: Would have to see how this would affect plotting/merging though. May complicate those.

**Comment by fabianvf** [comment]
1st pass done :+1:
If you have any questions about my comments, feel free to ask.

**Comment by bdyetton** [comment]
Deleting this PR cos the other branch has it covered

**Pull request closed without merging**